### PR TITLE
Add vagrant for dev env setup

### PR DESCRIPTION
**Pull request by snowch**
If the concept of vagrant is liked, it would be better to create a custom box using packer rather than rely on an upstream Vagrant box.

This would make provisioning much quicker. 

Adding this as a draft PR for discussion purposes.

### CODE REVIEW

File: vagrant/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant.configure("2") do |config|
+
+  config.vm.box = "mrvantage/centos7-minikube"
+  config.vm.network "forwarded_port", guest: 3000, host: 3000
+  config.vm.synced_folder "..", "/vagrant"
```
Comment (joel-bluedata) [inline vagrant/Vagrantfile:8]: cf. the discussion on issue #396 there could be an issue with this.

Comment (snowch) [reply vagrant/Vagrantfile]: Thanks, @joel-bluedata https://github.com/bluek8s/kubedirector/pull/403/commits/a31a6aab2923611f3a8e5f0f1ba9c0d92a2d1dca fixes #396

**Comment by snowch** [comment]
hi @joel-bluedata  - let me know if you would like me to walk you through the environment and also how it could be improved with packer

**Comment by snowch** [comment]
@joel-bluedata - I've come up with a better idea: the lab environment should be a separate repo that includes kubedirector as a git submodule.  This keeps the kubedirector repo clean and focussed.  

The new repo is here: https://github.com/hpe-container-platform-community/kubedirector-lab

**Comment by joel-bluedata** [comment]
Thanks Chris. I'm able to only intermittently check on stuff at the moment but I am keeping this on my radar.

**Pull request closed without merging**